core/src/main/java/com/es/core/model/phone/dao/JdbcColorDao.java

```
@@ -0,0 +1,81 @@
+package com.es.core.model.phone.dao;
+
+import com.es.core.model.phone.Color;
+import com.es.core.model.phone.util.ColorBatchPreparedStatementSetter;
+import jakarta.annotation.Resource;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Statement;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Repository
+public class JdbcColorDao implements ColorDao {
+
+    private static final String FROM_COLORS_WHERE_ID = "select * from colors where id = ?";
+    private static final String UPDATE_COLORS_SET_CODE_WHERE_ID = "update colors set code = ? where id = ?";
+    private static final String INSERT_INTO_COLORS_CODE_VALUES = "insert into colors (code) values (?)";
+    private static final String GET_SQL_WITH_OFFSET_AND_LIMIT = "select * from colors offset ? limit ?";
+    private static final String SELECT_FROM_COLORS_FOR_THE_PHONE = "select * from colors " +
+            "join phone2color pc on colors.id = pc.colorId " +
+            "where pc.phoneId = ?";
+
+    @Resource
+    private JdbcTemplate jdbcTemplate;
+
+    @Override
+    public Optional<Color> get(final Long id) {
+        if (id == null) {
+            return Optional.empty();
```

it's better to throw an exception when id is null (for example IllegalArgumentException) because it's incorrect when you pass null as id